index.js

```diff
@@ -57,9 +61,25 @@ function serialize(form, options) {
         if ((element.type === 'checkbox' || element.type === 'radio') && !element.checked) {
             val = undefined;
         }
+        
+        // If we want empty elements for check box
+        if (options.empty && element.type === 'checkbox' && !element.checked) {
+            val = "";
+        }
+
+        // If we want the empty elements for radio
+        if (options.empty && element.type === 'radio') {
+            if (!radio_store[element.name] && !element.checked) {
+                radio_store[element.name] = false
+            }
+            else if (element.checked) {
+                radio_store[element.name] = true
+            }
+        }
 
-        // value-less fields are ignored
-        if (!val) {
+        // value-less fields are ignored unless options.empty is true
+        // if options empty is true, continue only if its radio
+        if (!val && (!options.empty || (options.empty && element.type == "radio"))) {
```

Comment:
options.empty is checked several times in these three if statements (this one and two above). Can we do just a single if (options.empty)  ... else and do the logic for the two cases in those. Seems like that would be easier to follow.